### PR TITLE
Upgrade codeception version to 5

### LINKED ISSUE #11664: Update codeception/codeception to version 5
### Improvement description

Version 5 Supports Symfony >=4.4.24 <7.0

### PR DESCRIPTION
**Pull request by aryaantony92**
## Changes in this pull request  
Resolves #11664

### CODE REVIEW

File: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 1]
```diff
@@ -82,6 +82,7 @@ Please make sure to set your preferred storage location ***before*** migration.
 - [DataObjects] Removed deprecated preview url in class editor.
 - [DataObjects] Removed sql filter functionality for data object grid
 - [Device Detector] Bumped `matomo/device-detector` to ^6.0.
+- [Codeception] Bumped codeception version to 5.0
```
Comment (dvesh3) [inline doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]:
```suggestion
- [Codeception] Bumped `codeception/codeception` version to ^5.0
```

File: .github/workflows/codeception.yaml [hunk 2]
```diff
@@ -121,4 +121,4 @@ jobs:
                   ./bin/console debug:config flysystem
 
             - name: "Run Codeception"
-              run: "vendor/bin/codecept run -c . -vvv --json"
+              run: "vendor/bin/codecept run -c . -vvv --html"
```
Comment (dvesh3) [inline .github/workflows/codeception.yaml]: I would go for `--xml` option as it is easier to read on console.

File: codeception.dist.yml [hunk 3]
```diff
@@ -3,12 +3,13 @@ settings:
...
-actor: Tester
+support_namespace: Support
+actor_suffix: Tester
 paths:
     tests: tests
-    log: tests/_output
-    data: tests/_data
-    support: tests/_support
+    output: tests/_output
+    data: tests/Support/data
```
Comment (dvesh3) [inline codeception.dist.yml]:
```suggestion
    data: tests/Support/Data
```

File: doc/Development_Documentation/19_Development_Tools_and_Details/29_Testing/01_Application_Testing.md [hunk 4]
```diff
@@ -275,12 +275,13 @@ behaviour:
...
-actor: Tester
+support_namespace: Support
+actor_suffix: Tester
 paths:
     tests: .
-    log: ./_output
-    data: ./_data
-    support: ./_support
+    output: ./_output
+    data: ./Support/data
```
Comment (dvesh3) [inline doc/Development_Documentation/19_Development_Tools_and_Details/29_Testing/01_Application_Testing.md]:
```suggestion
    data: ./Support/Data
```

File: tests/Support/Helper/AbstractDefinitionHelper.php [hunk 5]
```diff
@@ -13,18 +13,18 @@
...
-use Pimcore\Tests\Util\TestHelper;
+use Pimcore\Tests\Support\Util\TestHelper;
 
 abstract class AbstractDefinitionHelper extends Module
 {
     /**
      * @var array
      */
-    protected $config = [
+    protected array $config = [
```
Comment (dvesh3) [inline tests/Support/Helper/AbstractDefinitionHelper.php:27]: remove unnecessary phpdocs

File: tests/Support/Helper/Pimcore.php [hunk 6]
```diff
@@ -361,4 +365,12 @@ public function makeHtmlSnapshot($name = null)
     {
         // TODO: Implement makeHtmlSnapshot() method.
     }
+
+    /**
+     * @return array
+     */
```
Comment (dvesh3) [inline tests/Support/Helper/Pimcore.php:371]:
```suggestion
```

File: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 7]
```diff
@@ -87,7 +87,7 @@ Please make sure to set your preferred storage location ***before*** migration.
 - Removed Pimcore Password Encoder factory, `pimcore_admin.security.password_encoder_factory` service and `pimcore.security.factory_type` config.
 - Removed BruteforceProtection
 - Removed PreAuthenticatedAdminToken
-- 
+- [Codeception] Bumped `codeception/codeception` version to ^5.0
```
Comment (dvesh3) [inline doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]: We should also mention that Pimcore is using new directory structure for tests, which breaks the existing implementations relying on the Pimcore test classes & support helpers.

Comment (aryaantony92) [reply doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]: Ok 👍

**Review by blankse — CHANGES_REQUESTED** [review]
Please update also the paths in .gitignore
https://github.com/pimcore/pimcore/blob/9d2088bb4c469a25a74caf84df00f87d2afb2a7d/.gitignore#L27-L30

**Comment by dvesh3** [comment]
> Please update also the paths in .gitignore
> 

CS Fixer too. https://github.com/pimcore/pimcore/blob/2f18a0f26380c56b609d937b40d2050c8d1a3e1a/.php-cs-fixer.dist.php#L13

**Review by dvesh3 — APPROVED** [review]

**Pull request merged**